src/components/Activities.jsx

```diff
@@ -1,79 +1,33 @@
-import { Component } from 'react';
+import { useState, useEffect } from 'react';
 import Activity from './Activity';
 import Poster from './Poster';
 import { KOALA_ACTIVITY_ENDPOINT } from '../helpers/env';
 import PropTypes from 'prop-types';
 
-export default class Activities extends Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      activities: [],
-    };
+export default function Activities({ current, onChange }) {
+  const activities = useActivities();
 
-    this.loadData();
-  }
-
-  //Utility function to change dates from activities to actual Date objects
-  setDate(activity) {
-    return Object.assign(
-      {},
-      activity,
-      { has_start_time: activity.start_date.indexOf('T') > -1 },
-      {
-        has_end_time: activity.end_date && activity.end_date.indexOf('T') > -1,
-      },
-      { start_date: new Date(activity.start_date) },
-      activity.end_date ? { end_date: new Date(activity.end_date) } : null
-    );
-  }
+  if (current >= activities.length - 1)
+    onChange(true);
 
-  // Get activities
-  loadData() {
-    fetch(KOALA_ACTIVITY_ENDPOINT)
-      // Fix activity dates and sort them on start_date
-      .then((resp) => resp.json())
-      .then((activities) =>
-        activities.map(this.setDate).sort((a, b) => a.start_date - b.start_date)
-      )
-      .then((activities) => {
-        this.setState({ activities: activities.filter((act) => act.poster) });
-      });
-  }
-
-  componentDidMount() {
-    // Set up interval.
-    // Every this.props.loadInterval, new events are loaded from koala.
-    this.dataLoader = setInterval(
-      this.loadData.bind(this),
-      parseInt(import.meta.env.VITE_LOAD_INTERVAL)
+  if (activities.length > 0) {
```

Comment:
Qua code style, you want to use 

```tsx
if(!activities.length) return <>  </>;

return <div> content </>; 
```

This will make following your logic easier. The same holds for where you use `if`, `else if` and `else`